component/s3storage/client.go

```diff
@@ -135,20 +149,67 @@ func (cl *Client) Configure(cfg Config) error {
 		})
 	}
 
-	// ListBuckets here to test connection
+	// ListBuckets here to test connection to S3 backend
 	bucketList, err := cl.ListBuckets()
 	if err != nil {
 		log.Err("Client::Configure : listing buckets failed. Here's why: %v", err)
+
+		var oe *smithy.OperationError
+		if errors.As(err, &oe) {
+			var re *awsHttp.ResponseError
+			if errors.As(err, &re) {
+				// Endpoint is invalid or could not connect to endpoint
+				if re.HTTPStatusCode() == http.StatusMovedPermanently || re.HTTPStatusCode() == 0 {
+					log.Err("Client::Configure : Error trying to connect to endpoint. Invalid endpoint. : %v", err)
+					return errInvalidEndpoint
+				}
+			}
+		}
+
+		var ae smithy.APIError
+		if errors.As(err, &ae) {
+			// If error is forbidden, then credentials were incorrect
+			if ae.ErrorCode() == "Forbidden" {
+				return errInvalidCredential
+			}
+			// If error is forbidden, then access key is correct but secret key is incorrect
+			if ae.ErrorCode() == "SignatureDoesNotMatch" {
+				return errInvalidSecretKey
+			}
+			fmt.Println(ae.Error())
+		}
 		return err
 	}
 
+	// If endpoint is localhost, then running locally so ignore region check
+	isLocalHost := strings.Contains(cl.Config.authConfig.Endpoint, "localhost")
+
+	// Check that region matches region in endpoint
+	// This check is after endpoint errors so endpoint can be validated first
+	if !strings.Contains(cl.Config.authConfig.Endpoint, cl.Config.authConfig.Region) && !isLocalHost {
+		log.Err("Endpoint in region does not match provided endpoint.")
+		return errRegionMismatch
+	}
```

Comment:
This would break when the endpoint has no region.
Can you use the error coming back from the cloud storage provider to identify this case instead?

Reply:
Good point, I moved this after we make a request.